### DISCUSSION TITLE
Mac M1芯片 secretflow1.9.0b2报错

### DISCUSSION THREAD

**Discussion by suk1M** [secretflow/secretflow#1535] [Q&A]
<img width="1136" alt="image" src="https://github.com/user-attachments/assets/665bf0c7-5c8f-45fd-80af-7344e7aba529"> 
请大佬们看看为什么报这个错  是少配置东西了吗

**Comment by wangzul** [comment]
这可能是因为权限问题、系统配置问题或程序本身的bug导致的。
1. 尝试执行uptime
这个命令会显示系统的运行时间以及过去的1分钟、5分钟和15分钟的平均负载。如果这个命令能正常输出，说明系统本身读取loadavg没有问题。
2. 查看系统日志（uptime正常失败，访问这个文件需要管理员权限）
/var/log/system.log
3. 尝试重启设备
4. 尝试重装sf
可以根据这些情况做一下验证。

**Reply by suk1M** [reply]
尝试完出现以下问题，烦请协助解决一下，万分感谢！
Since the GPL-licensed package `unidecode` is not installed, using Python's `unicodedata` package which yields worse results.
The version of SecretFlow: 1.9.0b2
Traceback (most recent call last):
  File "/Users/zyl/Workspace/pycharm/temp01/risk_control.py", line 28, in <module>
    spu = sf.SPU(sf.utils.testing.cluster_def(['alice', 'bob']))
  File "/Users/zyl/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/secretflow/device/device/spu.py", line 1549, in __init__
    self.init()
  File "/Users/zyl/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/secretflow/device/device/spu.py", line 1557, in init
    .remote(
  File "/Users/zyl/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/secretflow/distributed/memory_api.py", line 89, in remote
    mem_actor_handle.execute_impl(cls_args, cls_kwargs)
  File "/Users/zyl/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/secretflow/distributed/memory_actor.py", line 53, in execute_impl
    self._actor_handle = self._body(*cls_args, **cls_kwargs)
  File "/Users/zyl/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/secretflow/device/device/spu.py", line 377, in __init__
    self.link = spu_link.create_brpc(desc, rank)
RuntimeError: what: 
	[external/yacl/yacl/link/context.cc:195] connect to mesh failed, failed to setup connection to rank=1
stacktrace: 
#0 yacl::link::Context::ConnectToMesh()+0x16dbb9800
#1 pybind11::detail::argument_loader<>::call<>()+0x16bc0d4e0
#2 pybind11::cpp_function::initialize<>()::{lambda()#1}::__invoke()+0x16bc0d368
#3 pybind11::cpp_function::dispatcher()+0x16bbdf3fc
#4 cfunction_call+0x1043826dc
#5 _PyObject_MakeTpCall+0x10432f224
#6 call_function+0x104421618
#7 _PyEval_EvalFrameDefault+0x10441cf20
#8 _PyEval_Vector+0x104416320
#9 _PyObject_FastCallDictTstate+0x10432f4d8
#10 _PyObject_Call_Prepend+0x1043300dc
#11 slot_tp_init+0x1043a6f8c
#12 type_call+0x10439f954
#13 _PyObject_Call+0x10432fc38
#14 _PyEval_EvalFrameDefault+0x10441d718
#15 _PyEval_Vector+0x104416320


2024-10-17 09:41:20.817 [warning] [channel.h:~Channel:165] Channel destructor is called before WaitLinkTaskFinish, try stop send thread

Process finished with exit code 1


代码如下：

**Reply by suk1M** [reply]
// Check the version of your SecretFlow
print('The version of SecretFlow: {}'.format(sf.__version__))

// sf.shutdown()
// sf.init(['alice', 'bob'], address='local')
sf.init(['alice', 'bob'], debug_mode=True,)
alice, bob = sf.PYU('alice'), sf.PYU('bob')
spu = sf.SPU(sf.utils.testing.cluster_def(['alice', 'bob']))

**Comment by wangzul** [comment]
如果代码中有spu相关内容，由于spu的运行模式和同步模式不兼容，debug模式使用pyu进行了功能模拟，请务必使用simulation模式进行验证
[参考官网文档](https://www.secretflow.org.cn/zh-CN/docs/secretflow/v1.9.0b2/getting_started/debug_mode#id6)

**Reply by suk1M** [reply]
目前继续执行demo仍有报错，我自己查了一下 说是vdf参数的数据类型不对，烦请协助查看，谢谢！报错信息如下：
--------------------------------------------------------------------------
RayActorError                             Traceback (most recent call last)
Cell In[13], line 5
      2 from secretflow.preprocessing.binning.vert_bin_substitution import VertBinSubstitution
      4 binning = VertWoeBinning(spu)
----> 5 bin_rules = binning.binning(
      6     vdf,
      7     binning_method="chimerge",
      8     bin_num=4,
      9     bin_names={alice: [], bob: ["duration"]},
     10     label_name="y",
     11 )
     13 woe_sub = VertBinSubstitution()
     14 vdf, _ = woe_sub.substitution(vdf, bin_rules)

File ~/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/secretflow/preprocessing/binning/vert_woe_binning.py:221, in VertWoeBinning.binning(self, vdata, binning_method, bin_num, bin_names, label_name, positive_label, chimerge_init_bins, chimerge_target_bins, chimerge_target_pvalue, audit_log_path)
    216 label, label_holder_report = label_holder_worker.label_holder_work(
    217     vdata.partitions[label_holder_device].data
    218 )
    220 if isinstance(self.secure_device, SPU):
--> 221     secure_label = label.to(self.secure_device)
    222 elif isinstance(self.secure_device, HEU):
    223     secure_label = label.to(
    224         self.secure_device,
    225         HEUMoveConfig(heu_audit_log=label_holder_audit_log_path),
    226     )

File ~/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/secretflow/device/device/base.py:70, in DeviceObject.to(self, device, *args, **kwargs)
     60 def to(self, device: Device, *args, **kwargs):
     61     """Device object conversion.
     62 
     63     Args:
   (...)
     68         DeviceObject: Target device object.
     69     """
---> 70     return dispatch(_name_of_to(device.device_type), self, device, *args, **kwargs)

File ~/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/secretflow/device/device/register.py:111, in dispatch(name, self, *args, **kwargs)
    101 def dispatch(name: str, self, *args, **kwargs):
    102     """Dispatch device kernel.
    103 
    104     Args:
   (...)
    109         Kernel execution result.
    110     """
--> 111     return _registrar.dispatch(self.device_type, name, self, *args, **kwargs)

File ~/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/secretflow/device/device/register.py:80, in Registrar.dispatch(self, device_type, name, *args, **kwargs)
     78 if name not in self._ops[device_type]:
     79     raise KeyError(f'device: {device_type}, op: {name} not registered')
---> 80 return self._ops[device_type][name](*args, **kwargs)

File ~/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/secretflow/device/kernels/pyu.py:76, in pyu_to_spu(self, spu, spu_vis)
     73     ret = io.make_shares(data, vtype)
     74     return ret
---> 76 shares_chunk_count = self.device(get_shares_chunk_count)(
     77     self.data, spu.conf, spu.world_size, vtype
     78 )
     79 shares_chunk_count = sfd.get(shares_chunk_count.data)
     81 meta, io_info, *shares_chunk = self.device(
     82     run_spu_io, num_returns=(2 + shares_chunk_count * spu.world_size)
     83 )(self.data, spu.conf, spu.world_size, vtype)

File ~/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/secretflow/device/device/pyu.py:102, in PYU.__call__.<locals>.wrapper(*args, **kwargs)
     96 args_, kwargs_ = jax.tree_util.tree_map(
     97     lambda arg: try_get_data(arg, self), (args, kwargs)
     98 )
    100 _num_returns = check_num_returns(fn) if num_returns is None else num_returns
    101 data = (
--> 102     sfd.remote(self._run)
    103     .party(self.party)
    104     .options(num_returns=_num_returns)
    105     .remote(fn, *args_, **kwargs_)
    106 )
    107 logging.debug(
    108     (
    109         f'PYU remote function: {fn}, num_returns={num_returns}, '
    110         f'args len: {len(args)}, kwargs len: {len(kwargs)}.'
    111     )
    112 )
    113 if _num_returns == 1:

File ~/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/ray/remote_function.py:246, in RemoteFunction.options.<locals>.FuncWrapper.remote(self, *args, **kwargs)
    245 def remote(self, *args, **kwargs):
--> 246     return func_cls._remote(args=args, kwargs=kwargs, **updated_options)

File ~/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/secretflow/distributed/primitive.py:225, in RemoteFunctionWrapper._remote(self, *args, **kwargs)
    224 def _remote(self, *args, **kwargs):
--> 225     args, kwargs = _resolve_args(*args, **kwargs)
    226     return super()._remote(*args, **kwargs)

File ~/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/secretflow/distributed/primitive.py:215, in _resolve_args(*args, **kwargs)
    210 arg_flat, arg_tree = jax.tree_util.tree_flatten((args, kwargs))
    211 refs = {
    
    212     pos: arg for pos, arg in enumerate(arg_flat) if isinstance(arg, ray.ObjectRef)
    213 }
--> 215 actual_vals = ray.get(list(refs.values()))
    216 for pos, actual_val in zip(refs.keys(), actual_vals):
    217     arg_flat[pos] = actual_val

File ~/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/ray/_private/auto_init_hook.py:22, in wrap_auto_init.<locals>.auto_init_wrapper(*args, **kwargs)
     19 @wraps(fn)
     20 def auto_init_wrapper(*args, **kwargs):
     21     auto_init_ray()
---> 22     return fn(*args, **kwargs)

File ~/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/ray/_private/client_mode_hook.py:103, in client_mode_hook.<locals>.wrapper(*args, **kwargs)
    101     if func.__name__ != "init" or is_client_mode_enabled_by_default:
    102         return getattr(ray, func.__name__)(*args, **kwargs)
--> 103 return func(*args, **kwargs)

File ~/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/ray/_private/worker.py:2626, in get(object_refs, timeout)
   2624             raise value.as_instanceof_cause()
   2625         else:
-> 2626             raise value
   2628 if is_individual_id:
   2629     values = values[0]

RayActorError: The actor died because of an error raised in its creation task, ray::ActorVertWoeBinningPyuWorker.__init__() (pid=89872, ip=127.0.0.1, actor_id=c804bf10133d7b03ea8e20b901000000, repr=<secretflow.device.proxy.ActorVertWoeBinningPyuWorker object at 0x16a717a60>)
  File "/Users/zyl/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/secretflow/preprocessing/binning/vert_woe_binning_pyu.py", line 50, in __init__
    data_columns = data.column_names
  File "/Users/zyl/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/pandas/core/generic.py", line 5902, in __getattr__
    return object.__getattribute__(self, name)
AttributeError: 'DataFrame' object has no attribute 'column_names'

**Reply by suk1M** [reply]
![Uploading image.png…]()

**Reply by wangzul** [reply]
> ## 目前继续执行demo仍有报错，我自己查了一下 说是vdf参数的数据类型不对，烦请协助查看，谢谢！报错信息如下：
> RayActorError Traceback (most recent call last) Cell In[13], line 5 2 from secretflow.preprocessing.binning.vert_bin_substitution import VertBinSubstitution 4 binning = VertWoeBinning(spu) ----> 5 bin_rules = binning.binning( 6 vdf, 7 binning_method="chimerge", 8 bin_num=4, 9 bin_names={alice: [], bob: ["duration"]}, 10 label_name="y", 11 ) 13 woe_sub = VertBinSubstitution() 14 vdf, _ = woe_sub.substitution(vdf, bin_rules)
> 
> File ~/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/secretflow/preprocessing/binning/vert_woe_binning.py:221, in VertWoeBinning.binning(self, vdata, binning_method, bin_num, bin_names, label_name, positive_label, chimerge_init_bins, chimerge_target_bins, chimerge_target_pvalue, audit_log_path) 216 label, label_holder_report = label_holder_worker.label_holder_work( 217 vdata.partitions[label_holder_device].data 218 ) 220 if isinstance(self.secure_device, SPU): --> 221 secure_label = label.to(self.secure_device) 222 elif isinstance(self.secure_device, HEU): 223 secure_label = label.to( 224 self.secure_device, 225 HEUMoveConfig(heu_audit_log=label_holder_audit_log_path), 226 )
> 
> File ~/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/secretflow/device/device/base.py:70, in DeviceObject.to(self, device, *args, **kwargs) 60 def to(self, device: Device, *args, **kwargs): 61 """Device object conversion. 62 63 Args: (...) 68 DeviceObject: Target device object. 69 """ ---> 70 return dispatch(_name_of_to(device.device_type), self, device, *args, **kwargs)
> 
> File ~/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/secretflow/device/device/register.py:111, in dispatch(name, self, *args, **kwargs) 101 def dispatch(name: str, self, *args, **kwargs): 102 """Dispatch device kernel. 103 104 Args: (...) 109 Kernel execution result. 110 """ --> 111 return _registrar.dispatch(self.device_type, name, self, *args, **kwargs)
> 
> File ~/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/secretflow/device/device/register.py:80, in Registrar.dispatch(self, device_type, name, *args, **kwargs) 78 if name not in self._ops[device_type]: 79 raise KeyError(f'device: {device_type}, op: {name} not registered') ---> 80 return self._ops[device_type][name](*args, **kwargs)
> 
> File ~/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/secretflow/device/kernels/pyu.py:76, in pyu_to_spu(self, spu, spu_vis) 73 ret = io.make_shares(data, vtype) 74 return ret ---> 76 shares_chunk_count = self.device(get_shares_chunk_count)( 77 self.data, spu.conf, spu.world_size, vtype 78 ) 79 shares_chunk_count = sfd.get(shares_chunk_count.data) 81 meta, io_info, *shares_chunk = self.device( 82 run_spu_io, num_returns=(2 + shares_chunk_count * spu.world_size) 83 )(self.data, spu.conf, spu.world_size, vtype)
> 
> File ~/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/secretflow/device/device/pyu.py:102, in PYU.**call**..wrapper(*args, **kwargs) 96 args_, kwargs_ = jax.tree_util.tree_map( 97 lambda arg: try_get_data(arg, self), (args, kwargs) 98 ) 100 _num_returns = check_num_returns(fn) if num_returns is None else num_returns 101 data = ( --> 102 sfd.remote(self._run) 103 .party(self.party) 104 .options(num_returns=_num_returns) 105 .remote(fn, *args_, **kwargs_) 106 ) 107 logging.debug( 108 ( 109 f'PYU remote function: {fn}, num_returns={num_returns}, ' 110 f'args len: {len(args)}, kwargs len: {len(kwargs)}.' 111 ) 112 ) 113 if _num_returns == 1:
> 
> File ~/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/ray/remote_function.py:246, in RemoteFunction.options..FuncWrapper.remote(self, *args, **kwargs) 245 def remote(self, *args, **kwargs): --> 246 return func_cls._remote(args=args, kwargs=kwargs, **updated_options)
> 
> File ~/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/secretflow/distributed/primitive.py:225, in RemoteFunctionWrapper._remote(self, *args, **kwargs) 224 def _remote(self, *args, **kwargs): --> 225 args, kwargs = _resolve_args(*args, **kwargs) 226 return super()._remote(*args, **kwargs)
> 
> File ~/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/secretflow/distributed/primitive.py:215, in _resolve_args(*args, **kwargs) 210 arg_flat, arg_tree = jax.tree_util.tree_flatten((args, kwargs)) 211 refs = {
> 
> ```
> 212     pos: arg for pos, arg in enumerate(arg_flat) if isinstance(arg, ray.ObjectRef)
> 213 }
> ```
> 
> --> 215 actual_vals = ray.get(list(refs.values())) 216 for pos, actual_val in zip(refs.keys(), actual_vals): 217 arg_flat[pos] = actual_val
> 
> File ~/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/ray/_private/auto_init_hook.py:22, in wrap_auto_init..auto_init_wrapper(*args, **kwargs) 19 @wraps(fn) 20 def auto_init_wrapper(*args, **kwargs): 21 auto_init_ray() ---> 22 return fn(*args, **kwargs)
> 
> File ~/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/ray/_private/client_mode_hook.py:103, in client_mode_hook..wrapper(*args, **kwargs) 101 if func.**name** != "init" or is_client_mode_enabled_by_default: 102 return getattr(ray, func.**name**)(*args, **kwargs) --> 103 return func(*args, **kwargs)
> 
> File ~/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/ray/_private/worker.py:2626, in get(object_refs, timeout) 2624 raise value.as_instanceof_cause() 2625 else: -> 2626 raise value 2628 if is_individual_id: 2629 values = values[0]
> 
> RayActorError: The actor died because of an error raised in its creation task, ray::ActorVertWoeBinningPyuWorker.**init**() (pid=89872, ip=127.0.0.1, actor_id=c804bf10133d7b03ea8e20b901000000, repr=<secretflow.device.proxy.ActorVertWoeBinningPyuWorker object at 0x16a717a60>) File "/Users/zyl/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/secretflow/preprocessing/binning/vert_woe_binning_pyu.py", line 50, in **init** data_columns = data.column_names File "/Users/zyl/opt/anaconda3/envs/sf01/lib/python3.10/site-packages/pandas/core/generic.py", line 5902, in **getattr** return object.**getattribute**(self, name) AttributeError: 'DataFrame' object has no attribute 'column_names'

运行代码发一下。

**Comment by suk1M** [comment]
运行代码如上，烦请帮忙看看，谢谢！

**Comment by wangzul** [comment]
你的意思是现在在跑
【隐语SecretFlow金融风控全链路能力展示】示例代码并且没有使用debug_mode模式然后出现的错误。

**Comment by suk1M** [comment]
是的发自我的 iPhone在 2024年10月23日，10:14，wangzul ***@***.***> 写道：﻿
你的意思是现在在跑
【隐语SecretFlow金融风控全链路能力展示】示例代码并且没有使用debug_mode模式然后出现的错误。

**Reply by wangzul** [reply]
> 是的发自我的 iPhone在 2024年10月23日，10:14，wangzul ***@***.***> 写道：﻿ 你的意思是现在在跑 【隐语SecretFlow金融风控全链路能力展示】示例代码并且没有使用debug_mode模式然后出现的错误。 —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

执行一下pip list 我看一下环境

**Comment by suk1M** [comment]
请查收，谢谢！



> 2024年10月24日 16:02，wangzul ***@***.***> 写道：
> 
> 
> 是的发自我的 iPhone在 2024年10月23日，10:14，wangzul @.> 写道：﻿ 你的意思是现在在跑 【隐语SecretFlow金融风控全链路能力展示】示例代码并且没有使用debug_mode模式然后出现的错误。 —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: @.>
> 
> 执行一下pip list 我看一下环境
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/orgs/secretflow/discussions/1535#discussioncomment-11037792>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AHJUKPRB5462YQRYANQZR63Z5CSSBAVCNFSM6AAAAABP2HIK7KVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMBTG43TSMQ>.
> You are receiving this because you authored the thread.
> 


(sf01) ***@***.*** temp01 % pip list
Package                   Version
------------------------- --------------
absl-py                   2.1.0
aiobotocore               2.15.2
aiohappyeyeballs          2.4.3
aiohttp                   3.10.10
aioitertools              0.12.0
aiosignal                 1.3.1
anyio                     4.6.0
appnope                   0.1.4
argon2-cffi               23.1.0
argon2-cffi-bindings      21.2.0
arrow                     1.3.0
asttokens                 2.4.1
astunparse                1.6.3
async-lru                 2.0.4
async-timeout             4.0.3
attrs                     24.2.0
Babel                     2.14.0
beautifulsoup4            4.12.3
bleach                    6.1.0
botocore                  1.35.36
Brotli                    1.1.0
cached-property           1.5.2
cachetools                5.5.0
certifi                   2024.8.30
cffi                      1.17.1
charset-normalizer        3.4.0
clean-text                0.6.0
click                     8.1.7
cloudpickle               3.1.0
comm                      0.2.2
contourpy                 1.3.0
cycler                    0.12.1
debugpy                   1.8.7
decorator                 5.1.1
defusedxml                0.7.1
dill                      0.3.9
duckdb                    1.0.0
emoji                     1.7.0
entrypoints               0.4
exceptiongroup            1.2.2
executing                 2.1.0
fastjsonschema            2.20.0
filelock                  3.16.1
flatbuffers               2.0.7
fonttools                 4.54.1
fqdn                      1.5.1
frozenlist                1.4.1
fsspec                    2024.2.0
ftfy                      6.3.0
gast                      0.4.0
google-auth               2.35.0
google-auth-oauthlib      1.0.0
google-pasta              0.2.0
grpcio                    1.56.2
h11                       0.14.0
h2                        4.1.0
h5py                      3.12.1
hpack                     4.0.0
httpcore                  1.0.6
httpx                     0.27.2
hyperframe                6.0.1
idna                      3.10
importlib_metadata        8.5.0
importlib_resources       6.4.5
interconnection           0.2.0.dev0
ipykernel                 6.29.5
ipython                   8.28.0
ipywidgets                8.1.5
isoduration               20.11.0
jax                       0.4.26
jaxlib                    0.4.26
jedi                      0.19.1
Jinja2                    3.1.4
jmespath                  1.0.1
joblib                    1.4.2
json5                     0.9.25
jsonpointer               3.0.0
jsonschema                4.23.0
jsonschema-specifications 2024.10.1
jupyter                   1.1.1
jupyter_client            8.6.3
jupyter-console           6.6.3
jupyter_core              5.7.2
jupyter-events            0.10.0
jupyter-lsp               2.2.5
jupyter_server            2.14.2
jupyter_server_terminals  0.5.3
jupyterlab                4.2.5
jupyterlab_pygments       0.3.0
jupyterlab_server         2.27.3
jupyterlab_widgets        3.0.13
keras                     2.12.0
kiwisolver                1.4.7
kuscia                    0.0.3b0
libclang                  18.1.1
lightning-utilities       0.11.7
llvmlite                  0.40.1
Markdown                  3.7
MarkupSafe                3.0.1
matplotlib                3.9.2
matplotlib-inline         0.1.7
mistune                   3.0.2
ml_dtypes                 0.5.0
mpmath                    1.3.0
msgpack                   1.1.0
multidict                 6.1.0
multiprocess              0.70.17
munkres                   1.1.4
nbclient                  0.10.0
nbconvert                 7.16.4
nbformat                  5.10.4
nest_asyncio              1.6.0
networkx                  2.8.8
notebook                  7.2.2
notebook_shim             0.2.4
numba                     0.57.0
numpy                     1.23.5
oauthlib                  3.2.2
onnx                      1.16.0
opt_einsum                3.4.0
overrides                 7.7.0
packaging                 24.1
pandas                    1.5.3
pandocfilters             1.5.0
parso                     0.8.4
pexpect                   4.9.0
pickleshare               0.7.5
pillow                    10.4.0
pip                       24.2
pkgutil_resolve_name      1.3.10
platformdirs              4.3.6
prometheus_client         0.21.0
prompt_toolkit            3.0.48
propcache                 0.2.0
protobuf                  4.25.5
psutil                    6.0.0
ptyprocess                0.7.0
pure_eval                 0.2.3
pyarrow                   14.0.2
pyasn1                    0.6.1
pyasn1_modules            0.4.1
pycparser                 2.22
Pygments                  2.18.0
pyobjc-core               10.3.1
pyobjc-framework-Cocoa    10.3.1
pyparsing                 3.1.4
PySocks                   1.7.1
python-dateutil           2.9.0
python-json-logger        2.0.7
pytz                      2024.2
PyYAML                    6.0.2
pyzmq                     26.2.0
ray                       2.9.3
referencing               0.35.1
requests                  2.32.3
requests-oauthlib         2.0.0
rfc3339-validator         0.1.4
rfc3986-validator         0.1.1
rpds-py                   0.20.0
rsa                       4.9
s3fs                      2024.2.0
scikit-learn              1.3.2
scipy                     1.14.1
secretflow                1.9.0b2
secretflow-rayfed         0.2.1a2
secretflow-serving-lib    0.6.0b0
Send2Trash                1.8.3
setuptools                69.5.1
sf-heu                    0.5.0b0
six                       1.16.0
sniffio                   1.3.1
soupsieve                 2.5
sparse                    0.15.4
spu                       0.9.2b0
sqlglot                   25.5.1
stack-data                0.6.2
sympy                     1.13.3
tabulate                  0.9.0
tensorboard               2.12.3
tensorboard-data-server   0.7.2
tensorflow-estimator      2.12.0
tensorflow-macos          2.12.0
termcolor                 2.5.0
terminado                 0.18.1
tf2onnx                   1.14.0
threadpoolctl             3.5.0
tinycss2                  1.3.0
tomli                     2.0.2
torch                     2.1.2
torchaudio                2.1.2
torchmetrics              1.3.2
torchvision               0.16.2
tornado                   6.4.1
tqdm                      4.66.5
traitlets                 5.14.3
types-python-dateutil     2.9.0.20241003
typing_extensions         4.12.2
typing-utils              0.1.0
unicodedata2              15.1.0
uri-template              1.3.0
urllib3                   2.2.3
validator.py              1.3.0
wcwidth                   0.2.13
webcolors                 24.8.0
webencodings              0.5.1
websocket-client          1.8.0
Werkzeug                  3.0.4
wheel                     0.44.0
widgetsnbextension        4.0.13
wrapt                     1.14.1
xgboost                   1.7.5
yarl                      1.15.0
zipp                      3.20.2
zstandard                 0.23.0
(sf01) ***@***.*** temp01 %

**Comment by wangzul** [comment]
建议先使用secretflow 1.8.0b0，当前使用的示例代码比较老旧，需要更新。

**Comment by suk1M** [comment]
我现在用的1.9.0b2 ，版本不老旧吧 = =

**Reply by wangzul** [reply]
> 我现在用的1.9.0b2 ，版本不老旧吧 = =
> […](#)
> 2024年10月28日 10:17，wangzul ***@***.***> 写道： 建议先使用secretflow 1.8.0b0，当前使用的示例代码比较老旧，需要更新。 — Reply to this email directly, view it on GitHub <[#1535 (comment)](https://github.com/orgs/secretflow/discussions/1535#discussioncomment-11069921)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AHJUKPTMIDFR2EOUA3WB3ADZ5WNC7AVCNFSM6AAAAABP2HIK7KVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMBWHE4TEMI>. You are receiving this because you authored the thread.

错误提示是：AttributeError: 'DataFrame' object has no attribute 'column_names'
在secretflow=1.9版本中使用的是 data=pyarrow.Table
在secretflow=1.8版本中使用的是 data=pandas.DataFrame
在1.9版本中对于组件部分改动较大，按照现有的示例代码的运行会出现兼容性问题，因此建议先使用1.8版本运行。
后续会修复这个问题。
